src/main/java/de/hpi/bpt/argos/parsing/staticData/StaticDataParserImpl.java

```diff
@@ -0,0 +1,277 @@
+		if (staticDataDirectory.length() == 0) {
+			logger.info("static data directory not defined in properties");
+			return;
+		}
+
+		// since the path is delivered as URI, it is represented as a HTML string. Thus we need to replace %20 with file system spaces.
+		File dataDirectory = new File(staticDataDirectory.replaceAll("%20", " "));
+
+		if (!dataDirectory.isDirectory()) {
+			logger.error("static data directory is not a directory");
+			return;
+		}
+
+		File[] dataFiles = dataDirectory.listFiles();
+
+		if (dataFiles == null) {
+			logger.info("no static data files to parse");
+			return;
+		}
+
+		for (File staticDataFile : dataFiles) {
+			if (!staticDataFile.getName().endsWith(".xml")) {
+				continue;
+			}
+
+			WatchTask parseFileTask = null;
+
+			try {
+				int oldEntityTypesCount = PersistenceAdapterImpl.getInstance().getEntityTypesCount();
+				int oldEntitiesCount = PersistenceAdapterImpl.getInstance().getEntitiesCount();
+
+				parseFileTask = WatchImpl.start(String.format("parsing static data file: '%1$s'", staticDataFile.getName()),
+						() -> parse(staticDataFile));
+
+				int newEntityTypesCount = PersistenceAdapterImpl.getInstance().getEntityTypesCount();
+				int newEntitiesCount = PersistenceAdapterImpl.getInstance().getEntitiesCount();
+
+				logger.info(String.format("'%1$s' contained [%2$d entityTypes] and [%3$d entities]",
+						staticDataFile.getName(),
+						newEntityTypesCount - oldEntityTypesCount,
+						newEntitiesCount - oldEntitiesCount));
+
+			} catch (Exception e) {
+				LoggerUtilImpl.getInstance().error(logger, String.format("cannot parse '%1$s'", staticDataFile.getName()), e);
+
+			} finally {
+				if (parseFileTask != null) {
+					parseFileTask.finish();
+				}
+			}
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected void startElement(String elementName) {
+		switch (elementName) {
+			case ENTITY_TYPES_ROOT_ELEMENT:
+				activeSubParser = new EntityTypesParser(this, entityTypes);
+
+				if (activeWatchTask != null) {
+					activeWatchTask.finish();
+				}
+
+				activeWatchTask = WatchImpl.start("parsing entity types", () -> {});
+				return;
+
+			case ENTITY_INSTANCES_ROOT_ELEMENT:
+				activeSubParser = new EntityInstancesParser(this, entityTypes);
+
+				if (activeWatchTask != null) {
+					activeWatchTask.finish();
+				}
+
+				activeWatchTask = WatchImpl.start("parsing entity instances", () -> {});
+				return;
+
+			default:
+				break;
+		}
+
+		if (activeSubParser != null) {
+			activeWatchTask.run(() -> activeSubParser.startElement(elementName));
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected void element(String elementData) {
+		if (elementData.matches("^\\s*$")) {
+			return;
+		}
+
+		if (activeSubParser == null) {
+			return;
+		}
+
+		activeWatchTask.run(() -> activeSubParser.elementValue(latestOpenedElement(0), elementData));
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected void endElement(String elementName) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "endElement" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)